server/chat-commands/core.ts

```diff
@@ -654,7 +654,7 @@ export const commands: ChatCommands = {
 		if (target.length > maxLength) {
 			return this.errorReply(this.tr `Your status is too long; it must be under ${maxLength} characters.`);
 		}
-		target = this.statusfilter(target);
+		target = await this.statusfilter(target);
```

Comment:
I really don't think we should be propagating this.

We're adding a bunch of lag to all our commands for... what, exactly?

Reply:
Improved unit test capabilities? I'm not sure this is worth it. Can we frontload enough of autolock such that the important stuff just happens synchronously?

Reply:
We can't exactly frontload all the important stuff, but we can add an extra line to set `user.locked` before the full punishment codes runs, which i did in #7710 since you seem to prefer that approach. And yes, less fragile unit tests (not having to delay assertions by 25ms).